libr/core/cmd_anal.c

```diff
@@ -2779,6 +2780,21 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
 			r_core_cmdf (core, "\"afs %s\"", data);
 			free (sig);
 			free (data);
+		} else if (input[2] == 'r') { // "afsr"
+			if (input[3] == ' ') {
+				const char *ntype = ((input[4]) != '\0')
+					? (input + 4)
+					: NULL;
+				if (ntype) {
+					char *sig = r_core_cmd_str (core, "afs");
+					char *sig_notype = strchr (sig, ' ');
+					char *data = r_str_newf ("%s%s", ntype, sig_notype);
+					strtok (data, "\n");
+					r_core_cmdf (core, "\"afs %s\"", data);
```

Comment:
what about just passing %s%s in here instead of data?, the newline shouldnt be a problem if you filter the ntype before using it.